src/rpc/blockchain.cpp

```diff
@@ -1532,6 +1539,178 @@ UniValue getchaintxstats(const JSONRPCRequest& request)
+        } else if (plot_value == "avgfeerate") {
+            map_stats[plot_value].push_back(CFeeRate(total, total_weight).GetFee(1));
+        } else if (plot_value == "txs") {
+            map_stats[plot_value].push_back((int64_t)block.vtx.size());
+        } else if (plot_value == "ins") {
+            map_stats[plot_value].push_back(inputs);
+        } else if (plot_value == "outs") {
+            map_stats[plot_value].push_back(outputs);
+        }
+    }
+}
+
+static bool IsAllowedPlotValue(const std::string& plot_value, std::vector<std::string>& allowed_plot_values)
+{
+    for (const std::string& allowed_plot_value : allowed_plot_values) {
+        if (allowed_plot_value == plot_value) {
+            return true;
+        }
+    }
+    return false;
+}
+
+UniValue getperblockstats(const JSONRPCRequest& request)
+{
+    std::string str_allowed_plot_values = "minfee,maxfee,totalfee,minfeerate,maxfeerate,avgfee,avgfeerate,txs,ins,outs";
+    std::vector<std::string> allowed_plot_values;
+    boost::split(allowed_plot_values, str_allowed_plot_values, boost::is_any_of(","));
+
+    if (request.fHelp || request.params.size() < 1 || request.params.size() > 4)
+        throw std::runtime_error(
+            "getperblockstats ( nStart nEnd plotValues )\n"
+            "\nCompute per block statistics for a given window.\n"
+            "\nArguments:\n"
+            "1. \"start\"      (numeric, required) The height of the block that starts the window.\n"
+            "2. \"end\"        (numeric, optional) The height of the block that ends the window (default: current tip).\n"
+            "3. \"plotvalues\"  (string,  optional) Values to plot (comma separated), default(all): " + str_allowed_plot_values +
+            "\nResult:\n"
+            "{\n"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("getperblockstats", "1000 1000 \"minfeerate,avgfeerate\"")
+            + HelpExampleRpc("getperblockstats", "1000 1000 \"maxfeerate,avgfeerate\"")
+        );
+
+    LOCK(cs_main);
+
+    int start = request.params[0].get_int();
+    if (start < 1 || start > chainActive.Height()) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Start block height out of range");
+    }
+
+    int end;
+    if (request.params.size() > 1) {
+        end = request.params[1].get_int();
+    } else {
+        end = chainActive.Height();
+    }
+    if (end < 0 || end > chainActive.Height()) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "End block height out of range");
+    }
+
+    std::string str_plot_values = str_allowed_plot_values;
+    if (request.params.size() > 2) {
+        str_plot_values = request.params[2].get_str();
+    }
+    std::vector<std::string> plot_values;
+    boost::split(plot_values, str_plot_values, boost::is_any_of(","));
+    for (const std::string plot_value : plot_values) {
+        if (!IsAllowedPlotValue(plot_value, allowed_plot_values)) {
```

Comment:
duh, I was so much over-complicating things so much for no good reason...thank you!